### DISCUSSION TITLE
academy/puppeteer-playwright/page/interacting-with-a-page

### DISCUSSION THREAD

**Discussion by giscus** [apify/apify-docs#699] [Comments]
# academy/puppeteer-playwright/page/interacting-with-a-page

Learn how to programmatically do actions on a page such as clicking, typing, and pressing keys. Also, discover a common roadblock that comes up when automating.

https://docs.apify.com/academy/puppeteer-playwright/page/interacting-with-a-page

**Comment by danieluxury88** [comment]
I am testing the example and I get this error, after I commented out the click over "I agree" button (which does not appear on my case, I am on Ecuador). I also replaced the type selector from input to textarea:

@daniel-linux ➜ playwright-academy  node index.js
node:internal/process/esm_loader:46
      internalBinding('errors').triggerUncaughtException(
                                ^

page.type: Timeout 30000ms exceeded.
=========================== logs ===========================
waiting for locator('textarea[title="Search"]')
============================================================
    at file:///home/daniel/Projects/Apify/playwright-academy/index.js:13:12 {
  name: 'TimeoutError'
}

```
import { chromium } from 'playwright';

const browser = await chromium.launch({ headless: false });

const page = await browser.newPage();

await page.goto('https://google.com/');

// Click the "I agree" button
// await page.click('button:has-text("I agree")');

// Type the query into the search box
await page.type('textarea[title="Search"]', 'hello world');

// Press enter
await page.keyboard.press('Enter');

await page.waitForTimeout(10000);
await browser.close();
```

sorry I can help you more but still kind of new on web development.

**Reply by B4nan** [reply]
The problem with this guide is that it operates based on the text which differs based on your language settings, I can also see a different text on the button (since I am using different language than english in my browser). 

There is a PR trying to fix this guide here, be sure to check that out:

https://github.com/apify/apify-docs/pull/666
